patches/server/0355-Anti-Xray.patch

```diff
@@ -302,6 +302,7 @@ index 0000000000000000000000000000000000000000..3916d99c746d952e7ae6c3dde750d08f
 +    private final BlockState[] presetBlockStates;
 +    private final BlockState[] presetBlockStatesFull;
 +    private final BlockState[] presetBlockStatesStone;
++    private final BlockState[] presetBlockStatesDeepslate;
 +    private final BlockState[] presetBlockStatesNetherrack;
 +    private final BlockState[] presetBlockStatesEndStone;
 +    private final int[] presetBlockStateBitsGlobal;
```

Comment:
Unfortunately github doesn't allow to comment outside of the diff. Read the lines below this comment. This must also be implemented for deepslate, otherwise it will fail if a chunk section uses the global palette (this case is handled separately because it can be optimized).